xwiki-rendering-macros/xwiki-rendering-macro-box/src/main/java/org/xwiki/rendering/macro/box/AbstractBoxMacro.java

```diff
@@ -277,8 +278,12 @@ public Block build() throws MacroExecutionException
                 // we add the title, if there is one
                 if (!StringUtils.isEmpty(titleParameter)) {
                     // Don't execute transformations explicitly. They'll be executed on the generated content later on.
-                    ret.addChildren(AbstractBoxMacro.this.contentParser.parse(
-                        titleParameter, context, false, true).getChildren());
+                    List<? extends Block> titleBlock = AbstractBoxMacro.this.contentParser.parse(
+                        titleParameter, context, false, true).getChildren();
+
+                    // put the right metadata around it
+                    ret.addChildren(Collections.singletonList(new MetaDataBlock(titleBlock,
+                        AbstractBoxMacro.this.getNonGeneratedContentMetaData("title"))));
```

Comment:
Do you really need the ``AbstractBoxMacro.this.`` prefix here? It looks like the ``getNonGeneratedContentMetaData`` method is inherited from the base class.

Reply:
Actually I'm in an inner private class here, see: https://github.com/xwiki/xwiki-rendering/pull/173/files/dca4ddeddcd413aff36e6427212cba6af58b2d03#diff-0ae096925d6afadeedd83855fed76f80R190

Reply:
Right, sorry, I missed that.